app/controllers/pieces_controller.rb

```diff
@@ -19,7 +19,7 @@ def update
   private
   
     def set_piece
-      @piece = Piece.find(params[:id])
+      @piece ||= Piece.find(params[:id])
```

Comment:
Does this make sense? I feel like this method should only be run automatically (otherwise the `@piece` attribute is just used directly).